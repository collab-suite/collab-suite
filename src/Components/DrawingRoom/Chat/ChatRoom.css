* {
  margin: 0;
}

.chat-container {
}

/* Message Headers */

.msg-header-1 {
  float: left;
  color: #b5b5b5;
  font-size: 1rem;
  display: inline-block;
}

.msg-header-2 {
  float: right;
  color: #b5b5b5;
  font-size: 1rem;
  display: inline-block;
}


/* Messages */

.super1 {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.super2 {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

.msg1 {
  list-style: none;
  background: #72b8ff;
  border-radius: 20px;
  display: inline-block;
  padding: 10px;
  color: #fff;
  font-weight: lighter;
  font-size: small;
  box-shadow: 1px 1px 2px rgba(0,0,0,.3);
  margin: 5px;
  position: relative; 
}

.msg2 {
  list-style: none;
  background: #00D025;
  border-radius: 20px;
  display: inline-block;
  padding: 10px;
  color: #fff;
  font-weight: lighter;
  font-size: small;
  box-shadow: 1px 1px 2px rgba(0,0,0,.3);
  margin: 5px;
  position: relative; 
}

.msg1.u1 {
  float: left;
  clear: both;
  border-top-left-radius: 0px;
}

.msg1.u1:before {
  content: "";
  width: 0px;
  height: 0px;
  display: block;
  border-left: 5px solid transparent;
  border-right: 5px solid #72b8ff;
  border-top: 5px solid #72b8ff;
  border-bottom: 5px solid transparent;
  position: absolute;
  top: 0px;
  left: -10px;
}

.msg2.u2 {
    float: right;
    clear: both;
    border-top-right-radius: 0px;
}

.msg2.u2:before {
  content: "";
  width: 0px;
  height: 0px;
  display: block;
  border-left: 5px solid #00D025;
  border-right: 5px solid transparent;
  border-top: 5px solid #00D025;
  border-bottom: 5px solid transparent;
  position: absolute;
  top: 0px;
  right: -10px;
}

/* Message Form */

.msg-form {
  width: 100%;
  position: absolute;
  bottom: 10px;
  display: flex;
  flex-direction: column;
}

/* Message Input */

.msg-input {
  margin: 10px 5px;
  outline: none;
  border-radius: 5px;
  padding: 10px;
  /* box-sizing: border-box; */
  font-size: 15px;
  width: 14.8%;
  height: 40px;
  border: none;
  /* display: inline-block; */
  color: #0f0f0f;
  font-weight: 100;
  background: #ededed;
  /* position: bottom; */
}

/* Message Buttons */

.join-room-btn-container {
  display: flex;
  justify-content: center;
}

.msg-btn {
  width: 4.5%;
  height: 40px;
  padding: 0;
  display: inline-block;
  border: none;
  color: #fffffc;
  background: #11b3c1;
  position: relative;
  outline: none;
  cursor: pointer;
  font-size: 1rem;
  margin-left: 5px;
  border-radius: 50px;
}

.msg-btn:hover {
  background: #fffffc;
  color: #11b3c1;
  border: solid 2px #11b3c1;
}

.join-room-btn {
  border: none;
  outline: none;
  font-size: 1rem;
  margin-top: 1rem;
}

.join-room-btn:hover {
  color: #11b3c1;
}
