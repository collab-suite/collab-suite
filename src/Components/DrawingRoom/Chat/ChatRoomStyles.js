import styled from 'styled-components'

export const ChatContainer = styled.div`
    height: 10%;
    width: 40%;
`

export const ChatRoom = styled.ul`
    list-style: none;
`
export const Message = styled.li`
    padding: 0;
    max-width: 100%;
    display: flex;
    justify-content: space-between;
`

export const Name = styled.h4`
    width: 30%;
    margin: 0;
    display: flex;
`

export const MessageText = styled.p`
    margin: 0;
    display: flex;
    max-width: 70%;
`


export const MessageForm = styled.form`

`

export const ChatInput = styled.input`
    border: none;
    outline: none;
`

export const SendMessage = styled.button`

`