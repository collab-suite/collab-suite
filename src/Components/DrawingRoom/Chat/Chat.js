import React, {useState, useEffect} from 'react'
import {useSelector} from 'react-redux'
import {withRouter} from 'react-router-dom'
import swal from '@sweetalert/with-react'
import './ChatRoom.css'
import axios from 'axios';

function Chat(props) {
    const user = useSelector(reduxState => reduxState.user)
    const [messages, setMessages] = useState([])
    const [usersDisp, setUsersDisp] = useState([])
    const [chatMessage, setChatMsg] = useState('')
    const {socket} = props
    let onlineUsers = usersDisp.map((ele, i) => {
        return (
            <li key={i}>{ele.first_name} {ele.last_name.charAt(0)}</li>
        )
    })
    function showUsersOnline() {
        return (
            <ul>
                {onlineUsers}
            </ul>
        )
    }
    useEffect(() => {
        socket.on('message recieved', messages => setMessages(messages))
        socket.on('joined room', (messages, users) => {
            setMessages(messages)
            setUsersDisp(users)
        })
    }, [])
    function showLink() {
        return swal({
            text: `${user.roomID}`
        })
    }
    function showOnline() {
        return swal({
            content: showUsersOnline
        })
    }
    function handleSendMessage() {
        let userInfo = {
            message: chatMessage,
            firstName: user.firstName,
            lastName: user.lastName,
            roomID: user.roomID
        }
        socket.emit('message send', userInfo)
        setChatMsg('')
    }
    function leaveRoom() {
        axios.delete(`/rooms/user?email=${user.email}`)
        .then(res => {
            swal({
                title: 'Left room',
                icon: 'success'
            })
            props.history.push('/')
        })
    }
    function endRoom() {
        axios.delete(`/rooms/end?roomID=${user.roomID}`)
        .then(res => {
            swal({
                title: 'Ended room',
                icon: 'success'
            })
            props.history.push('/')
        })
    }
    console.log(usersDisp)
    const messageDisplay = messages.map((ele, i) => {
        if (ele.first_name === user.firstName && ele.last_name === user.lastName) {
            return (
                <div className='super1' key={i}>
                    <h4 className='msg-header-1'>{ele.first_name} {ele.last_name.charAt(0)} </h4>
                    <li className='u1 msg1'>
                        <p className='msg-text'>{ele.message}</p>
                    </li>
                </div>
            )
        } else {
            return (
                <div className='super2' key={i}>
                    <h4 className='msg-header-2'>{ele.first_name} {ele.last_name.charAt(0)} </h4>
                    <li className='u2 msg2' key={i}>
                        <p className='msg-text-2'>{ele.message}</p>
                    </li>
                </div>
            )
        }
    })
    return (
        <div className='chat-container'>
            <div className='join-room-btn-container'>
                <button onClick={showLink} className='join-room-btn'>Join Room</button>
                {user.createdRoom?
                    <button className='join-room-btn' onClick={endRoom}>End Room</button>
                :
                    <button className='join-room-btn' onClick={leaveRoom}>Leave Room</button>
                }
                <button className='join-room-btn' onClick={showOnline}>Show Users</button>
            </div>
            <ul className='dialogue'>
                {messageDisplay}
            </ul>
            <div className='msg-form-container'>
                <form className='msg-form' onSubmit={e => e.preventDefault()}>
                    <textarea className='msg-input' placeholder='Type your message' value={chatMessage} onChange={e => setChatMsg(e.target.value)} />
                    <button className='msg-btn' onClick={handleSendMessage}>Send</button>
                </form>
            </div>
        </div>
    )
}

export default withRouter(Chat)