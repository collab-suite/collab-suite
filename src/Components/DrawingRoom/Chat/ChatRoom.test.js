import React from 'react';
import { shallow, mount, render} from 'enzyme';
import Chat from './Chat.js';
import configureStore from 'redux-mock-store'



describe('Canvas', () => {
    const initialState = {}
    const mockStore = configureStore()
    let store, container

  beforeEach(() => {
      store = mockStore(initialState)
      container = shallow(<Chat store={store} />)
})

  it('should render correctly in "debug" mode', () => {
  
    expect(container.length).toMatchSnapshot();
  });

  it('should render correctly in "debug" mode', () => {
    expect(container.length).toEqual(1)
  });
  it('should load a canvas css property', () => {
    expect(container.find('.cool-stuff-bro')).toHaveLength(1);
});
});