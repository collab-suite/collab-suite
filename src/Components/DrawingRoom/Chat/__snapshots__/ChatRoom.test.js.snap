// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canvas should render correctly in "debug" mode 1`] = `1`;
